### ISSUE TITLE
Subnet CRD stuck in deleted state

### ISSUE THREAD

**Issue by aminmr** [kubeovn/kube-ovn#3169]
# Expected Behavior
After uninstalling the helm chart, everything was deleted completely.

# Actual Behavior
The subnets.kubeovn.io CRD is always stuck and must be manually deleted by erasing the finalizer.

# Steps to Reproduce the Problem

1. Instal the helm chart 
2. Uninstall the helm chart
3. `kubect get crd | grep ovn`

**Note: This problem has been tested via Helm chart v1.12.0**

# Additional Info
![image](https://github.com/kubeovn/kube-ovn/assets/61911987/547d92dd-9482-4d96-ac01-36bb4c0e10a9)

By deleting this CRD, it will be stuck in a deleted state. Only by erasing the finalizer in the manifest will the CRD be deleted.

![image](https://github.com/kubeovn/kube-ovn/assets/61911987/ba085a48-7f31-4b7b-8f96-3c5063716fa7)

- Kubernetes version:
`1.25.6`

  **Output of `kubectl version`:**

```  
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.4", GitCommit:"fa3d7990104d7c1f16943a67f11b154b71f6a132", GitTreeState:"archive", BuildDate:"2023-07-20T07:37:53Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6", GitCommit:"ff2c119726cc1f8926fb0585c74b25921e866a28", GitTreeState:"clean", BuildDate:"2023-01-18T19:15:26Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
```
- **kube-ovn version**:
`1.12.0`

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
```
"Debian GNU/Linux 11 (bullseye)"
```
  **Output of `uname -r`:**
```
5.10.0-25-amd64
```

**Comment by zhangzujian** [comment]
I cannot find a way to fix it currently. Any ideas will be appreciated.

**Comment by zhangzujian** [comment]
Helm hook:

Annotation Value | Description
-- | --
pre-delete | Executes on a deletion request before any resources are deleted from Kubernetes

**Comment by aminmr** [comment]
hello @zhangzujian 
If you need any help, I would appreciate to contribute.

**Comment by zhangzujian** [comment]
> hello @zhangzujian If you need any help, I would appreciate to contribute.

Any contribution is appreciated. You can create a PR when your patch is ready.

**Comment by zhangzujian** [comment]
This issue will be fixed in v1.12.1.

Currently you need to execute the following command before uninstalling the helm release:

```sh
for subnet in $(kubectl get subnet -o name); do
  kubectl patch "$subnet" --type='json' -p '[{"op": "replace", "path": "/metadata/finalizers", "value": []}]'
done
```
